src/main/java/appeng/util/Platform.java

```diff
@@ -1431,17 +1431,17 @@ public static void configurePlayer( final EntityPlayer player, final AEPartLocat
 		player.rotationYaw = player.prevCameraYaw = player.cameraYaw = yaw;
 	}
 
-	public static boolean canAccess( final AENetworkProxy gridProxy, final BaseActionSource src )
+	public static boolean canAccess( final AENetworkProxy gridProxy, final IActionSource src )
 	{
 		try
 		{
-			if( src.isPlayer() )
+			if( src.player().isPresent() )
 			{
-				return gridProxy.getSecurity().hasPermission( ( (PlayerSource) src ).player, SecurityPermissions.BUILD );
+				return gridProxy.getSecurity().hasPermission( src.player().get(), SecurityPermissions.BUILD );
 			}
-			else if( src.isMachine() )
+			else if( src.machine().isPresent() )
 			{
-				final IActionHost te = ( (MachineSource) src ).via;
+				final IActionHost te = src.machine().get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3655)